types/prosemirror-view/index.d.ts

```diff
@@ -627,6 +627,16 @@ export interface EditorProps<ThisT = unknown, S extends Schema = any> {
    */
   scrollMargin?: number | { top: number, right: number, bottom: number, left: number } | null;
 }
+/**
+ * A mapping of dom events.
+ */
+export type HandleDOMEventsProp<ThisT = unknown, S extends Schema = any> = Partial<
+  {
+    [K in keyof DocumentEventMap]: (this: ThisT, view: EditorView<S>, event: DocumentEventMap[K]) => boolean;
+  }
+> & {
+  [key: string]: (this: ThisT, view: EditorView<S>, event: any) => boolean;
```

Comment:
Optional: You can simplify this a bit:

```typescript
{
    [K in keyof DocumentEventMap]?: (this: ThisT, view: EditorView<S>, event: DocumentEventMap[K]) => boolean;
} & { /* ... */ }
```

Reply:
I take it you mean I should remove the `[key: string]` part?

Reply:
No. You can replace the `Partial` wrapper by adding a `?` after `[K in keyof DocumentEventMap]`.